truss/templates/control/control/helpers/truss_patch/requirement_name_identifier.py

```diff
@@ -1,10 +1,40 @@
+import logging
+import re
 from typing import Dict, List
+from urllib.parse import parse_qs, urlparse
 
 from packaging import requirements  # type: ignore
 
+logger: logging.Logger = logging.getLogger(__name__)
+
+URL_PATTERN = re.compile(r"^(https?|git|svn|hg|bzr)\+")
+
+
+def is_url_based_requirement(req: str) -> bool:
+    return bool(URL_PATTERN.match(req))
+
 
 def identify_requirement_name(req: str) -> str:
     try:
+        # parse as a url requirement
```

Comment:
we don't have to do this now, but a refactoring that could be helpful here is instead of retruning a string, returning some richer type, that has the "no egg" information.

then the caller above doesn't have to duplicate this logic, and can just read that there's no egg and do the logging. no need for duplicate logging either

Reply:
gave it a try - LMK!